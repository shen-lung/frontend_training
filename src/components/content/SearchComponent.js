import React, { Component } from 'react';
import PropTypes from 'prop-types';


export default class SearchComponent extends Component {
    static  propTypes = {
        imageUrl: PropTypes.string.isRequired,
        eventName: PropTypes.string.isRequired,
        startEventDate: PropTypes.string.isRequired,
        eventPlace: PropTypes.string.isRequired,
        startEventPrice: PropTypes.string.isRequired,
    }
    render() {
        let {
            imageUrl,
            eventName,
            startEventDate,
            eventPlace,
            startEventPrice
        } = this.props;
        let imageContent;
        let price = 'Starts at ';

        if(!startEventPrice.includes('Free')) {
            if(startEventPrice.includes('Sold out')) {
                price = 'Sold out';
            } else {
                price = price + startEventPrice;
            }
            imageContent = (
                <div className="search-result__visual">
                    <img src={imageUrl} data-spec="event_image" />
                </div>
            );
        } else {
            price = 'Free';
            imageContent = (
                <div className="search-result__visual">
                    <img src={imageUrl} />
                    <div className="search-result__visual__sticker">
                        <p data-spec="free_sticker">Free</p>
                    </div>
                </div>
            );
        }

        return (
            <div>
                <div className="search-result">
                    {imageContent}
                    <div className="search-result__event-desc">
                        <h2 className="search-result__event-desc__title" data-spec="event_name">{eventName}</h2>
                        <div className="search-result__event-desc__info">
                            <p data-spec="starts_event_date">{startEventDate}</p>
                            <p data-spec="event_place">{eventPlace}</p>
                            <p data-spec="start_event_price">{price}</p>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
