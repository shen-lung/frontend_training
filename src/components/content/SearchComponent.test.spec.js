import React from 'react';
import {mount} from 'enzyme';

import SearchComponent from './SearchComponent';

const baseProps = {
    imageUrl: 'src/img/dj.jpg',
    eventName: 'That Event DJ - Stylish DJs for Oakstop',
    startEventDate: 'Sat, Aug 11, 11:00am',
    eventPlace: 'Oakstop, Oakland, CA',
    startEventPrice: '$67.50',
}

describe('Search Component', () => {
    it('should render', () => {
        const component = mount(<SearchComponent {...baseProps} />);

        expect(component).toMatchSnapshot();
    });

    it('should render event name', () => {
        const component = mount(<SearchComponent {...baseProps} />);

        expect(component.find('[data-spec="event_name"]')).toHaveText('That Event DJ - Stylish DJs for Oakstop');
    });

    it('should render starts event date', () => {
        const component = mount(<SearchComponent {...baseProps} />);

        expect(component.find('[data-spec="starts_event_date"]')).toHaveText('Sat, Aug 11, 11:00am');
    });

    it('should render event place', () => {
        const component = mount(<SearchComponent {...baseProps} />);

        expect(component.find('[data-spec="event_place"]')).toHaveText('Oakstop, Oakland, CA');
    });

    it('should render start event price', () => {
        const component = mount(<SearchComponent {...baseProps} />);

        expect(component.find('[data-spec="start_event_price"]')).toHaveText('Starts at $67.50');
    });

    it('should render event image', () => {
        const component = mount(<SearchComponent {...baseProps} />);

        expect(component.find('[data-spec="event_image"]').prop('src')).toEqual('src/img/dj.jpg');
    });

    it('should render sticker free', () => {
        let props = {...baseProps, startEventPrice: 'Free'};
        const component = mount(<SearchComponent {...props} />);

        expect(component.find('[data-spec="free_sticker"]').exists()).toBe(true);
    });
});
