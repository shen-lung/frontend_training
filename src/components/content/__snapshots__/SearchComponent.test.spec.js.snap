// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render 1`] = `
<SearchComponent
  eventName="That Event DJ - Stylish DJs for Oakstop"
  eventPlace="Oakstop, Oakland, CA"
  imageUrl="src/img/dj.jpg"
  startEventDate="Sat, Aug 11, 11:00am"
  startEventPrice="$67.50"
>
  <div>
    <div
      className="search-result"
    >
      <div
        className="search-result__visual"
      >
        <img
          data-spec="event_image"
          src="src/img/dj.jpg"
        />
      </div>
      <div
        className="search-result__event-desc"
      >
        <h2
          className="search-result__event-desc__title"
          data-spec="event_name"
        >
          That Event DJ - Stylish DJs for Oakstop
        </h2>
        <div
          className="search-result__event-desc__info"
        >
          <p
            data-spec="starts_event_date"
          >
            Sat, Aug 11, 11:00am
          </p>
          <p
            data-spec="event_place"
          >
            Oakstop, Oakland, CA
          </p>
          <p
            data-spec="start_event_price"
          >
            Starts at 
            $67.50
          </p>
        </div>
      </div>
    </div>
  </div>
</SearchComponent>
`;
